modules/administration/pages/custom-ssl.adoc

```diff
@@ -1,97 +1,195 @@
+--set-state="STATE" --set-city="CITY" --set-org="ORGANIZATION" \
+--set-org-unit="ORGANIZATION UNIT" --set-email="name@example.com" \
+--set-hostname="susemanager.example.top" --set-cname="example.com"
+----
+Ensure that the [systemitem]``set-cname`` parameter is the fully-qualified domain name of your {productname} Server.
+You can use the the [systemitem]``set-cname`` parameter multiple times if you require multiple aliases.
+. Generate the CA certificate RPM:
++
+----
+rhn-ssl-tool --gen-ca --rpm-only --dir="/root/ssl-build" \
+--from-ca-cert=`path_to_certificate_file`
+----
+. Generate the server certificate RPM:
++
+----
+rhn-ssl-tool --gen-server --rpm-only --dir="/root/ssl-build" \
+--from-server-key=`path_to_server_key_file` \
+--from-server-cert=`path_to_server certificate_file`
+----
+You will need to generate a server certificate RPM for each proxy, using their host names and cnames.
+. Create a new CA file that combines the old and new certificate details, and generate a new RPM:
++
+----
+mkdir /root/combined-ssl-build
+cp /root/old-ssl-build/RHN-ORG-TRUSTED-SSL-CERT /root/combined-ssl-build/
+cat /root/ssl-build/RHN-ORG-TRUSTED-SSL-CERT >> /root/combined-ssl-build/RHN-ORG-TRUSTED-SSL-CERT
+cp /root/old-ssl-build/*.rpm /root/combined-ssl-build/
+rhn-ssl-tool --gen-ca --rpm-only --dir="/root/combined-ssl-build"
+----
+// I would like to split up these steps, I think. LKB 2019-09-10
+. Deploy the CA certificate on the server:
++
+----
+/usr/bin/rhn-deploy-ca-cert.pl --source-dir /root/combined-ssl-build \
+--target-dir /srv/www/htdocs/pub/ --trust-dir=/etc/pki/trust/anchors/
+----
+
+When you have completed this procedure, you can deploy the new certificate on your clients.
 
-. Execute [command]``configure-proxy.sh``.
-. When prompted with:
+
+.Procedure: Deploying Certificates on Traditional Clients
+
+. On the client, create a new custom channel using these details:
 +
+* Name: SSL-CA-Channel
+* Label: ssl-ca-channel
+* Parent Channel: <choose the parent channel of a clients>
+* Summary: SSL-CA-Channel
 
++
+For more on creating custom channels, see xref:administration:channel-management.adoc[].
+. Upload the CA certificate RPM to the channel:
++
 ----
-Do you want to import existing certificates?
+rhnpush -c ssl-ca-channel --nosig \
+--ca-chain=/srv/www/htdocs/pub/RHN-ORG-TRUSTED-SSL-CERT \
+/root/combined-ssl-build/rhn-org-trusted-ssl-cert-1.0-2.noarch.rpm
 ----
-+
-Answer with kbd:[y]
-.
-. Continue by following the script prompts.
+. Subscribe all clients to the new ``SSL-CA-Channel`` channel.
+. Install the CA certificate RPM on all clients by updating the channel.
+
+.Procedure: Deploying Certificates on Salt Clients
+
+// Does the above procedure also work for Salt? LKB 2019-09-10
+
+. On the client, in the {productname} {webui}, navigate to menu:Salt[Remote Commands].
+. In the first field, enter [guimenu]``salt-call state.apply certs``.
+In the second field, after the ``@``, enter [guimenu]``*``.
+Click btn:[Find Targets].
```

Comment:
This is correct when you want to just deploy the new CA, but maybe we should just try out if a `highstate` on salt clients would do the same. I checked and the "certs" state is part of the highstate.

Reply:
Yeah, I was wondering about that.

Reply:
I think we should change it and say 

. In the {productname} {webui} navigate to the Systems Overview Page and select all Salt Managed Clients into System Set Manager (SSM)
. Navigate to Systems => System Set Manager => Overview and select  `Apply system states`
. Apply the Highstate on all systems in the system set manager by clicking on the "Apply Highstate" button